### PR TITLE
adds ansible role support

### PR DESCRIPTION
**Pull request by LarsFronius**
This is to aid unifying scripts with Codeship Basic.

This adds new scripts and changes some up to accommodate upcoming changes to the Basic deployment.
The old script entry point `codeship_heroku` this repo offered is made backward compatible _only_ for the case it's called with `deploy` as the first parameter.
A new script `codeship_heroku_deployment` was added, which should be the new entry point for both Basic and Pro.

This repo now also comes with an Ansible role, that is being tested here (decoupled from Basic), which is imported as a role into the Basic build.

### CODE REVIEW

File: Dockerfile.ansible [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM ubuntu:14.04
```
Comment (LarsFronius) [inline Dockerfile.ansible:1]: adds a dockerfile using ansible, to test the ansible role that checkbot will eventually pull in

File: codeship-services.yml [hunk 2]
```diff
@@ -1,12 +1,19 @@
+heroku_on_ansible:
```
Comment (LarsFronius) [inline codeship-services.yml:1]: builds mentioned container using ansible to configure itself.

File: codeship-steps.yml [hunk 3]
```diff
@@ -1,9 +1,24 @@
...
   service: heroku
-  command: codeship_heroku deploy /app/test codeship-deployment-test
+  command: codeship_heroku_deploy -f /app -N codeship-test-ruby-app
+- name: Deployment Test via plain container with old entry point/command
+  service: heroku
+  command: codeship_heroku deploy /app codeship-test-ruby-app
 - name: Heroku Run Test
   service: heroku
-  command: heroku run --exit-code --app codeship-deployment-test -- true
+  command: heroku run --exit-code --app codeship-test-ruby-app -- true
```
Comment (LarsFronius) [inline codeship-steps.yml:21]: changing up the used app to a sandbox app, which fixes https://github.com/codeship-library/heroku-deployment/issues/18

File: codeship-steps.yml [hunk 4]
```diff
@@ -13,7 +28,7 @@
     - name: Test Push
       service: dockercfg_test
       type: push
-      image_name: registry.heroku.com/codeship-deployment-test/web
+      image_name: registry.heroku.com/codeship-container-deploy-test/web
```
Comment (LarsFronius) [inline codeship-steps.yml:31]: also moves to a sandbox app to fix https://github.com/codeship-library/heroku-deployment/issues/18

File: deployment/scripts/codeship_heroku [hunk 5]
```diff
@@ -10,10 +12,15 @@ SCRIPT_NAME=${1:?'You need to provide the action as the first parameter'}
 
 SCRIPT_PATH=$DIR/codeship_heroku_$SCRIPT_NAME
 
-if [ -x $SCRIPT_PATH ]
-then
-  $SCRIPT_PATH "${@:2}"
+if [ -x "${SCRIPT_PATH}" ]; then
+  if [ "${SCRIPT_NAME}" == "deploy" ]; then
```
Comment (LarsFronius) [inline deployment/scripts/codeship_heroku:16]: Deprecating the codeship_heroku script for anything other than "deploy". "deploy" calls the new deployment script, matching old params against new ones.

File: deployment/scripts/codeship_heroku_deploy [hunk 6]
```diff
@@ -1,62 +1,126 @@
...
+-c or CODESHIP_HEROKU_CHECK_APP_URL: check url if service is running after deployment
+-h: this
 
-deployment_status=`echo "$deployment_result_json" | jq -r .status`
-echo "DEPLOYMENT STATUS: $deployment_status"
+EOF
+}
 
-[ "$deployment_status" = "succeeded" ]
+function check_url {
```
Comment (LarsFronius) [inline deployment/scripts/codeship_heroku_deploy:35]: check_url is a simplified version of the check_url script we have in Basic.

File: deployment/test/web.rb [hunk 7]
```diff
@@ -1,5 +1,7 @@
 require 'sinatra'
 
+
 get '/' do
-  "Hello, world"
+  whom = File.open("whom.txt").read
```
Comment (LarsFronius) [inline deployment/test/web.rb:5]: Planning to use this in the deployment test, to actually test a string on the site.

File: tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,52 @@
+- name: install heroku apt key
```
Comment (LarsFronius) [inline tasks/main.yml:1]: Ansible role we'll pull into the Basic container and is tested here with the ansible container.

**Comment by mlocher** [comment]
I'm 👍 on the changes.

**Pull request merged**